app/javascript/packages/analytics/click-observer-element.ts

```diff
@@ -0,0 +1,34 @@
+import { trackEvent } from '.';
+
+class ClickObserverElement extends HTMLElement {
+  trackEvent: typeof trackEvent = trackEvent;
+
+  connectedCallback() {
+    this.addEventListener('click', () => this.handleClick());
+  }
+
+  get eventName(): string | null {
+    return this.getAttribute('event-name');
+  }
+
+  /**
+   * Logs an event using the element's given event name.
+   */
+  handleClick() {
+    if (this.eventName) {
+      this.trackEvent(this.eventName);
```

Comment:
just confirming, this won't stop/wait for the tracking to be successful before proceeding right?

Reply:
> just confirming, this won't stop/wait for the tracking to be successful before proceeding right?

Yes, that's correct, and it can be a problem if the user were to be navigated away from the page as a result of whatever click is being tracked. It's not a problem for how we're currently using this, but it could be worth keeping an eye on. 

For example, there's a similar use-case here, which isn't using this element at the moment:

https://github.com/18F/identity-idp/blob/e170866f6dbc14f707ce63b2920551f316af96fd/app/javascript/packages/document-capture/components/in-person-prepare-step.tsx#L39-L40